### ISSUE TITLE
Custom callbacks are broken with Keras 2.4.Z wheels

### ISSUE THREAD

**Issue by Lothiraldan** [keras-team/keras#14134]
While upgrading to Keras 2.4.Z, I got a weird Traceback when using custom callbacks:

```
Using TensorFlow backend.
Callback defined in keras.callbacks.callbacks
60000 train samples
10000 test samples
2020-06-24 15:06:02.820522: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2020-06-24 15:06:02.820546: E tensorflow/stream_executor/cuda/cuda_driver.cc:313] failed call to cuInit: UNKNOWN ERROR (303)
Traceback (most recent call last):
  File "keras_example.py", line 74, in <module>
    main()
  File "keras_example.py", line 41, in main
    train(x_train, y_train, x_test, y_test)
  File "keras_example.py", line 68, in train
    x_train, y_train, batch_size=120, epochs=10, validation_data=(x_test, y_test), callbacks=[MyCallback()]
  File ".../python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 66, in _method_wrapper
    return method(self, *args, **kwargs)
  File ".../python3.7/site-packages/tensorflow/python/keras/engine/training.py", line 826, in fit
    steps=data_handler.inferred_steps)
  File ".../python3.7/site-packages/tensorflow/python/keras/callbacks.py", line 231, in __init__
    cb._implements_train_batch_hooks() for cb in self.callbacks)
  File ".../python3.7/site-packages/tensorflow/python/keras/callbacks.py", line 231, in <genexpr>
    cb._implements_train_batch_hooks() for cb in self.callbacks)
AttributeError: 'MyCallback' object has no attribute '_implements_train_batch_hooks'
```

I'm using latest tensorflow version, 2.22.0.

I digged a bit and found that it's not happening with latest Keras version installed from sdist. I installed it using the following pip command: `pip install --no-binary ":all:" -U keras`.

```
Callback defined in tensorflow.python.keras.callbacks
60000 train samples
10000 test samples
2020-06-24 15:08:15.681460: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory
2020-06-24 15:08:15.681477: E tensorflow/stream_executor/cuda/cuda_driver.cc:313] failed call to cuInit: UNKNOWN ERROR (303)
Epoch 1/10
```

It seems that the Keras wheel also includes a `callbacks` package that is imported in place of the new module https://github.com/keras-team/keras/blob/master/keras/callbacks.py.

```
$> unzip -l Keras-2.4.2-py2.py3-none-any.whl
       41  06-17-2020 23:03   keras/callbacks.py
      714  09-17-2019 17:12   keras/callbacks/__init__.py
    45641  10-07-2019 20:07   keras/callbacks/callbacks.py
    16934  09-17-2019 17:12   keras/callbacks/tensorboard_v1.py
     5447  09-17-2019 17:12   keras/callbacks/tensorboard_v2.py
```

This callbacks package doesn't seems present in the sdist package:

```
$> tar -tvf Keras-2.4.2.tar.gz | grep callbacks
-rw-r--r-- fchollet/primarygroup   41 2020-06-18 01:03 Keras-2.4.2/keras/callbacks.py
drwxr-xr-x root/primarygroup         0 2020-06-19 06:32 Keras-2.4.2/tests/keras/callbacks/
-rw-r--r-- fchollet/primarygroup 35586 2020-06-18 01:03 Keras-2.4.2/tests/keras/callbacks/callbacks_test.py
-rw-r--r-- fchollet/primarygroup 10186 2020-06-18 01:03 Keras-2.4.2/tests/keras/callbacks/tensorboard_test.py
drwxr-xr-x root/primarygroup          0 2020-06-19 06:32 Keras-2.4.2/docs/site/callbacks/
-rw-r--r-- fchollet/primarygroup  38830 2019-09-18 00:22 Keras-2.4.2/docs/site/callbacks/index.html
-rw-r--r-- fchollet/primarygroup   18342 2019-09-18 00:17 Keras-2.4.2/docs/sources/callbacks.md
-rw-r--r-- fchollet/primarygroup    2234 2019-08-22 18:24 Keras-2.4.2/docs/templates/callbacks.md
```

For the users that might be impacted, you can add replace the keras line in your requirements.txt file by:
```
keras --no-binary=keras
```

It also works with a version specifier:

```
keras>2.4.0 --no-binary=keras
```

**Comment by fchollet** [comment]
Interesting catch. There was also a mismatch with backend objects. This is now fixed. It was a file cache issue.

**Comment by Lothiraldan** [comment]
Thanks for the prompt fix, I tested locally both the 2.4.3 wheel and sdist and they work fine!
